src/Compilers/Shared/ShadowCopyAnalyzerAssemblyLoader.cs

```diff
@@ -49,6 +48,9 @@ public ShadowCopyAnalyzerAssemblyLoader(string baseDirectory = null)
                 _baseDirectory = Path.Combine(Path.GetTempPath(), "CodeAnalysis", "AnalyzerShadowCopies");
             }
 
+            _shadowCopyDirectoryAndMutex = new Lazy<(string directory, Mutex)>(
```

Comment:
>Lazy [](start = 47, length = 4)

What's the benefit for the `Lazy` here, compared to doing `_shadowCopyDirectoryAndMutex = CreateUniqueDirectoryForProcess();` in `LoadImpl(...)` method (where the `string` and `Mutex` where previously getting initialized)?

Reply:
LoadImpl might get called from multiple threads at once, but we want to have one directory and Mutex to cover it. The use of the Mutex here is so other processes can ask "is this directory still being used by some other process?" so we avoid deleting files out from underneath the owner. If you overwrite your previous directory/mutex, you're leaving those files open for collection by another process.